js/ui/overview.js

```diff
@@ -523,6 +526,41 @@ const Overview = new Lang.Class({
         this.hide();
     },
 
+    toggleWindows: function() {
+        if (this.isDummy)
+            return;
+
+        if (!this.visible) {
+            this.showWindows();
+            return;
+        }
+
+        if (!Main.workspaceMonitor.hasActiveWindows) {
+            // TODO: Show a dialog with a 'No apps open' message to the user.
+            return;
+        }
+
+        if (this.viewSelector.getActivePage() !== ViewSelector.ViewPage.WINDOWS) {
+            this.showWindows();
+            return;
+        }
+
+        if (!this._toggleToHidden) {
```

Comment:
Actually, no... further testing while addressing your comments made me realize I actually need that, together with the remaining code not present here which I removed. Will add it back